app/services/audit/AuditService.scala

```diff
@@ -121,17 +122,36 @@ class AuditService @Inject()(connector: AuditConnector, @Named("appName") appNam
       )
     )
 
-  def auditAllPagesUserInput(choice: Choice, userInput: JsObject)(implicit hc: HeaderCarrier): Future[AuditResult] = {
+  def auditAllPagesUserInput(choice: Choice, cacheMap: CacheMap)(implicit hc: HeaderCarrier): Future[AuditResult] = {
     val auditType = choice.value
     val extendedEvent = ExtendedDataEvent(
       auditSource = appName,
       auditType = auditType,
       tags = getAuditTags(s"${auditType}-payload-request", s"${auditType}/full-payload"),
-      detail = getAuditDetails(userInput)
+      detail = getAuditDetails(getMovementsData(choice, cacheMap))
     )
     connector.sendExtendedEvent(extendedEvent).map(handleResponse(_, auditType))
   }
 
+  def getMovementsData(choice: Choice, cacheMap: CacheMap): JsObject = {
+    val movementDetails =
+      if (choice.value == Choice.AllowedChoiceValues.Arrival)
+        Json.toJson(cacheMap.getEntry[ArrivalDetails](MovementDetails.formId))
+      else Json.toJson(cacheMap.getEntry[DepartureDetails](MovementDetails.formId))
+
+    val userInput = Map(
+      ConsignmentReferences.formId -> Json.toJson(
+        cacheMap.getEntry[ConsignmentReferences](ConsignmentReferences.formId)
+      ),
+      GoodsDeparted.formId -> Json.toJson(cacheMap.getEntry[GoodsDeparted](GoodsDeparted.formId)),
+      Location.formId -> Json.toJson(cacheMap.getEntry[Location](Location.formId)),
+      MovementDetails.formId -> movementDetails,
+      Transport.formId -> Json.toJson(cacheMap.getEntry[Transport](Transport.formId)),
+      ArrivalReference.formId -> Json.toJson(cacheMap.getEntry[ArrivalReference](ArrivalReference.formId))
+    )
+    Json.toJson(userInput).as[JsObject]
```

Comment:
Is it `as[JsObject]` necessary? You'll have `JsObject` directly from `Json.toJson(...)`

Reply:
This has been addressed.

Reply:
Actually you need it, otherwise, you get a compilation error